ExampleMod/Projectiles/ExampleBobber.cs

```diff
@@ -0,0 +1,157 @@
+using ExampleMod.Items;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+using static Terraria.ModLoader.ModContent;
+
+namespace ExampleMod.Projectiles
+{
+	public class ExampleBobber : ModProjectile
+	{
+		// You can use vanilla textures by using the format: Terraria/Item_<ID>
+		public override string Texture => "Terraria/Projectile_" + ProjectileID.BobberWooden;
+		private bool initialized = false;
+		private int color = 0;
+		private static Color fishingLineColor = new Color(255, 215, 0);
+
+		public override void SetStaticDefaults() {
+			DisplayName.SetDefault("Example Bobber");
+		}
+
+		public override void SetDefaults() {
+			//These are copied through the CloneDefaults method
+			//projectile.width = 14;
+			//projectile.height = 14;
+			//projectile.aiStyle = 61;
+			//projectile.bobber = true;
+			//projectile.penetrate = -1;
+			projectile.CloneDefaults(ProjectileID.BobberWooden);
+		}
+
+		//What if we want to randomize the line color
+		public override void AI() {
+			if (!initialized) {
+				color = Main.rand.Next(2);
+				initialized = true;
+			}
+			//50% chance for the line to be blue instead of the default gold
+			if (color == 1) {
+				fishingLineColor = new Color(0, 191, 255);
+			}
+		}
+
+		public override bool PreDrawExtras(SpriteBatch spriteBatch) {
+			//Create some bluish light; this could also be in the AI function
+			Lighting.AddLight(projectile.Center, 0f, 0.45f, 0.46f);
+
+			//Change these two values in order to change the origin of where the line is being drawn
+			int xPositionAdditive = 45;
+			float yPositionAdditive = 35f;
+
+			Player player = Main.player[projectile.owner];
+			if (projectile.bobber && player.inventory[player.selectedItem].holdStyle > 0) {
+				float originX = player.MountedCenter.X;
+				float originY = player.MountedCenter.Y;
+				originY += player.gfxOffY;
+				int type = player.inventory[player.selectedItem].type;
+				//This variable is used to account for Gravitation Potions
+				float gravity = player.gravDir;
+
+				if (type == ItemType<ExampleFishingRod>()) {
+					originX += (float)(xPositionAdditive * player.direction);
+					if (player.direction < 0) {
+						originX -= 13f;
+					}
+					originY -= yPositionAdditive * gravity;
+				}
+
+				if (gravity == -1f) {
+					originY -= 12f;
+				}
+				Vector2 mountedCenter = new Vector2(originX, originY);
+				mountedCenter = player.RotatedRelativePoint(mountedCenter + new Vector2(8f), true) - new Vector2(8f);
+				float projPosX = projectile.Center.X - mountedCenter.X;
```

Comment:
I assume this is all the vanilla code for calculating the fishing line segments. Some general comments would be nice on method (parabolic, weight modelling, whatever it is) but not necessary.

I would however like you to change all the X and Y variable pairs to Vector2 from XNA

Reply:
I think it is vanilla code.  I actually got it from another modder and cleaned it up myself with some help.  I decided to finally make an Example Fishing Rod because I remembered it didn't exist when I wanted to make my own and had line drawing issues before I got this code.

My knowledge of code is not enough to understand half of what is happening in the draw code.